### PR TITLE
feat(frontend): support create table with connector syntax

### PR DESCRIPTION
**Pull request by xx01cyx**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

###  Summarize your change 

Support syntax:
```sql
CREATE TABLE t (
  v int
) WITH (
  connector = 'xxx',
  ...
) ROW FORMAT xxx;
```

This is equivalent to `CREATE MATERIALIZED SOURCE`.

### Describe any limitations of the current code

- `DROP TABLE` may not work as expected. This will be fixed after unsupporting `CREATE MATERIALIZED SOURCE`.
- No test cases for the new syntax. This will be done after unsupporting `CREATE MATERIALIZED SOURCE` (so I could rename all `CREATE MATERIALIZED SOURCE` to `CREATE TABLE`).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

#5949

### CODE REVIEW

**Comment by TennyZhuang** [comment]
Is this a user-facing change?

**Comment by xx01cyx** [comment]
> Is this a user-facing change?

Yes. But this feature has not been completed yet, for there exist some limits (such as the user cannot perform DML on table created with a connector now).

**Review by st1page — COMMENTED** [review]
generally LGTM, PTAL @bohan

**Review by tabVersion — APPROVED** [review]
LGTM. good work

File: src/frontend/src/handler/create_source.rs [hunk 1]
```diff
@@ -227,25 +200,71 @@ pub async fn handle_create_source(
...
-                csv_delimiter: csv_info.delimiter as i32,
-                csv_has_header: csv_info.has_header,
+                row_format: RowFormatType::CanalJson as i32,
                 ..Default::default()
-            },
-        ),
+            }
+        }
+
+        SourceSchema::CSV(csv_info) => StreamSourceInfo {
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:210]: Would you mind also change `CSV` to `Csv` in this PR? 😂

**Review by BugenZhao — COMMENTED** [review]
rubber stamp

**Pull request merged**